### ISSUE TITLE
Bug: use_fallback = true causing single-repeat issues

### ISSUE THREAD

**Issue by EvanEzell** [VonHeikemen/lsp-zero.nvim#366]
When I do not set `use_fallback = false` (the default setting), I am able to change an inner word in normal mode (ciw followed by my change) and successively repeat the command on other words with the '.' single-repeat normal mode command. However, when I set `use_fallback = true`, I can no longer successively repeat commands. I can successfully repeat the command once after changing an inner word, but it fails to repeat the command more than one time. I have tried to debug this and have narrowed it down to the `s.try_complete_fallback()` function inside of `omnifunc.lua` [here](https://github.com/VonHeikemen/lsp-zero.nvim/blob/2d4b39bfbfd27d16db24af4ea566f81610207d69/lua/lsp-zero/omnifunc.lua#L189).

```lua
function M.autocomplete_fallback()
  augroup(group, {clear = false})

  autocmd('InsertCharPre', {
    group = group,
    desc = 'Autocomplete using words in current file',
    callback = s.try_complete_fallback,
  })
end

function s.try_complete_fallback()
  if pumvisible() > 0 or s.is_macro() or s.is_prompt() then
    return
  end

  if match(vim.v.char, pattern) >= 0 then
    vim.api.nvim_feedkeys(buffer_code, 'n', false)
  end
end
``` 

I think that the buffer_code being passed to nvim through `vim.api.nvim_feedkeys(buffer_code, 'n', false)` is somehow messing up the redo-register. Looking at the documentation of [vim.api.nvim_feedkeys](https://neovim.io/doc/user/api.html#nvim_feedkeys()) it states that this function is subject to various quirks. I couldn't find any further details. More broadly, it is likely that this issue applies to any situation where you want to repeat a previous change and that change involved typing characters in insert mode (triggering the fallback code).

**Comment by VonHeikemen** [comment]
Should be fixed with [dec1c21](https://github.com/VonHeikemen/lsp-zero.nvim/commit/dec1c21204e2d9d49dad989b577c88958ed2c113)

Update lsp-zero and let me know if it worked.

**Comment by EvanEzell** [comment]
This fixes the issue. Thank you!

Would you be willing to explain further the changes you made as I could not figure out exactly what was going on myself? I see now that you have added an additional keymap that translates your custom strings (`plug_omni = '<Plug>(lsp-zero-complete-omni)'` and `plug_buffer = '<Plug>(lsp-zero-complete-word)'`) to the previous key codes ('<C-x><C-o>' and '<C-x><C-n>') when in insert mode. How does this extra layer of translation solve the issue?

**Comment by VonHeikemen** [comment]
Honestly, I don't know why it works. My guess is the "key code" generated by the `<Plug>` keymap doesn't have the same bug as the one generated by `<C-x><C-o>`.

Could not find anything useful on google so I read the source code of [vim-simple-complete](https://github.com/maxboisvert/vim-simple-complete). That's the first autocomplete plugin I used, back when I started learning about vim. They have that `<Plug>` keymap trick, and it seems to work for them, so that's why I tried here.
